pkg/k8shandler/podtemplate.go

```diff
@@ -0,0 +1,90 @@
+package k8shandler
+
+import (
+	"reflect"
+
+	"github.com/openshift/elasticsearch-operator/pkg/utils"
+	"github.com/openshift/elasticsearch-operator/pkg/utils/comparators"
+	v1 "k8s.io/api/core/v1"
+)
+
+// ArePodTemplateSpecDifferent compares two v1.PodTemplateSpecs
+// and returns True or False
+func ArePodTemplateSpecDifferent(lhs, rhs v1.PodTemplateSpec) bool {
+
+	return ArePodSpecDifferent(lhs.Spec, rhs.Spec, true)
+}
+
+// Abstracted logic into comparing pod specs so that we can check if our change has been rolled out
+// yet or not
+func ArePodSpecDifferent(lhs, rhs v1.PodSpec, strictTolerations bool) bool {
+	changed := false
+
+	if len(lhs.Containers) != len(rhs.Containers) {
+		changed = true
+	}
+
+	// check nodeselectors
+	if !areSelectorsSame(lhs.NodeSelector, rhs.NodeSelector) {
+		//logrus.Debugf("Resource '%s' has different nodeSelector than desired", node.self.Name)
+		changed = true
+	}
+
+	// strictTolerations are for when we compare from the deployments or statefulsets
+	// if we are seeing if rolled out pods contain changes we don't want strictTolerations
+	//   since k8s may add additional tolerations to pods
+	if strictTolerations {
+		// check tolerations
+		if !areTolerationsSame(lhs.Tolerations, rhs.Tolerations) {
+			//logrus.Debugf("Resource '%s' has different tolerations than desired", node.self.Name)
+			changed = true
+		}
+	} else {
+		// check tolerations
+		if !areLhsTolerationsContainedInRhs(lhs.Tolerations, rhs.Tolerations) {
+			//logrus.Debugf("Resource '%s' has different tolerations than desired", node.self.Name)
+			changed = true
+		}
+	}
+
+	// check container fields
+	for _, lContainer := range lhs.Containers {
+		found := false
+
+		for _, rContainer := range rhs.Containers {
+			// Only compare the images of containers with the same name
+			if lContainer.Name == rContainer.Name {
```

Comment:
No need to check for volume mounts?

Reply:
strangely we didn't check this before... lets open a bz to track that external to this... this pr will at least make it so we only have to add a check in one place